### PR TITLE
Add Bank of Monsec challenge

### PR DESCRIPTION
**Pull request by lduck11007**
1. Bank of Monsec
2. pwn
3. 3-4?
4. Check the features enabled and for any vulnerable functions, or any functions being used incorrectly.
5. The format might be difficult, but _leaking_ something may make your job easier.
6. No
7. Neither

### CODE REVIEW

**Comment by mariuszskon** [comment]
Please do something with git so that you don't re-add the easy_oracle challenge in this PR.

**Review by mariuszskon — CHANGES_REQUESTED** [review]
Congratulations, you have made a surprisingly difficult challenge if it wasn't for the NX bit being off :)

You have made a very nice writeup too :+1:

1. ROP is difficult (impossible?) because you happened to use a default flag for `fcf-protection` which I did not even know exists  (you can read up on it here: https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html)
2. Format string attacks are made significantly more difficult due to `FORTIFY_SOURCE` (https://man7.org/linux/man-pages/man7/feature_test_macros.7.html)

Ubuntu has an overview of some default options: https://wiki.ubuntu.com/ToolChain/CompilerFlags

This is fine btw, I'm just letting you know of things you might not have ;)

There is a pressing issue however.
The binary in the container and the one provided do not match.

Given binary sha256sum
```
3c6279d3e5b96ee080a438bc2090792a40b8fbb08ff77ad406a4f30f32160537  bankofmonsec
```

The one within docker

```
d2d6fa0e6e37ae9aa667c0d2655406a2431aebe3c6e6953b2f301bcab16d48dd  bankofmonsec
```
I suggest using `docker cp` to get the right binary out of the container so you can publish it on git. Alternatively, you can simply copy the binary from the host to the container.
I think the binary within the container won't have `fcf-protection` (unless if you add the flag, unless if it wasn't implemented back in 18.04) but don't quote me on that.

File: Submissions/bank_of_monsec/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM ubuntu:18.04
+RUN dpkg --add-architecture i386
+RUN apt-get update && apt-get install -y libc6:i386 build-essential socat gcc-multilib
+
+RUN useradd -ms /bin/bash user
+COPY . /bankofmonsec
```
Comment (mariuszskon) [inline Submissions/bank_of_monsec/Dockerfile:6]: This copies a lot more than it should into the container - when someone pwns it, they get the writeup and solution script and the Dockerfile... this is not ideal.

**Review by mariuszskon — CHANGES_REQUESTED** [review]
Challenge looks good now!
Just a minor thing to the Dockerfile and we are good to merge.

File: Submissions/bank_of_monsec/Dockerfile (hunk 1 above)
Comment (lduck11007) [reply Submissions/bank_of_monsec/Dockerfile:6]: I coped the dockerfile from Daniel's example submission, I'll change it in a second

**Review by mariuszskon — APPROVED** [review]
LGTM

**Pull request merged**